### DISCUSSION TITLE
Mega Menu : Ability to choose Content Width for each mega menu item

### DISCUSSION THREAD

**Discussion by Michaelgimii** [elementor/elementor#22078] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Hi team

Currently, the Content Width option sets the mega menu width for all items uniformly. **For example, we cannot have two Dropdown Content in a menu widget, one full width, and the other fit to content.**

### Describe the solution you'd like

I think it would be much better if you remove the Content Width option from its current location and add this option for each item (in the same section where we enable Dropdown Content). **In this case, we can set the Content Width for each item separately, and that's great.**

**Comment by DavidGMiles** [comment]
I second this - the current all-or-nothing approach is very inflexible

**Comment by k4znIm** [comment]
Agree, this feature is missing. It would be much more flexible if we can decide for each one separately.

**Comment by 79ho3ein** [comment]
This is really needed. Of course, this is possible now with some tricks. But this feature should be available by default. I hope this feature will be added soon.

**Comment by DavidGMiles** [comment]
Do you mind sharing the trick as to how you would do it currently - as I do have  desperate need for this

**Reply by 79ho3ein** [reply]
Of course @DavidGMiles


https://github.com/elementor/elementor/assets/77163248/dedee250-d09f-4032-adb3-25d918362be5

**Reply by DavidGMiles** [reply]
> Of course @DavidGMiles
> 
Thanks - not like mine but it got me started and then a long way along - I needed some custom javascript in mine too to relocated things

I ended up with this

https://natureslens.co.uk/mega-menu-test/#

It still has some foibles and mobile view is bloody useless - so I turned it off

**Comment by lensbreak** [comment]
+1
